### PR TITLE
Inserting validation for categories names in --categories option when the ml-git entity create command is used

### PR DESCRIPTION
**Pull request by talessatiro**
Depends on: https://github.com/HPInc/ml-git/pull/164

This PR validates if the categories names (related with --categories option) have a valid Git tag name when the user executes the ml-git entity create command.

If an invalid category name is passed, the following message will be displayed on terminal:

image

### CODE REVIEW

File: ml_git/commands/custom_types.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    6. They cannot contain a sequence @{.
+    7. They cannot be the single character @.
+    8. They cannot contain a \.
+    """
+
+    name = 'git tag name'
+
+    def convert(self, value, param, ctx):
+        tag_name = super().convert(value, param, ctx)
+        valid_tag_name_regex = r'^(?!\/|@)((?!\/{2,}|\.{2,}|@{)(?=[^[^?*:\\])[(-}])+(?<!\.lock)(?<![/.])$'
```
Comment (gilMars) [inline ml_git/commands/custom_types.py:61]: move the regex expression to constants.py

Comment (talessatiro) [reply ml_git/commands/custom_types.py]: Changed on https://github.com/HPInc/ml-git/pull/169/commits/8bc7a817d417afd3c1788aa5d2119e2c9e49e78b

**Review by gilMars — APPROVED** [review]

File: ml_git/constants.py [hunk 2]
```diff
@@ -43,6 +43,7 @@
 RGX_SIZE_FILES = r'[+]\s+size:\s+(\d+(?:[.]\d+)*\s+.+)'
```
Comment (lucasncabral) [inline ml_git/constants.py:43]: Update header.

**Review by lucasncabral — APPROVED** [review]

File: ml_git/constants.py (hunk 2 above)
Comment (talessatiro) [reply ml_git/constants.py:43]: Changed on https://github.com/HPInc/ml-git/pull/169/commits/59295481cc64a5de2af0559dc71aa4add8450ae2

File: tests/integration/test_19_create.py [hunk 3]
```diff
@@ -246,7 +246,7 @@ def test_18_create_datasets_with_multiple_categories(self):
             self.assertEqual(spec_file[DATASET_SPEC_KEY]['categories'], ['cat1', 'cat2', 'cat3'])
 
     @pytest.mark.usefixtures('switch_to_tmp_dir')
-    def test_18_create_labels_with_multiple_categories(self):
+    def test_19_create_labels_with_multiple_categories(self):
```
Comment (andersonpablosilva) [inline tests/integration/test_19_create.py:249]: Please, fix the sequence for the previous tests.

**Review by andersonpablosilva — CHANGES_REQUESTED** [review]

File: tests/integration/test_19_create.py (hunk 3 above)
Comment (talessatiro) [reply tests/integration/test_19_create.py]: Changed on https://github.com/HPInc/ml-git/pull/169/commits/a0989ff4838de9b9399835f53c03f3d429e4eee9

**Review by andersonpablosilva — APPROVED** [review]

**Pull request merged**